controlplane/rosa/controllers/rosacontrolplane_controller.go

```diff
@@ -0,0 +1,387 @@
+		Namespace string
+		RoleARN   string
+		Path      string
+	}{
+		{
+			Name:      "cloud-credentials",
+			Namespace: "openshift-ingress-operator",
+			RoleARN:   rosaScope.ControlPlane.Spec.RolesRef.IngressARN,
+		},
+		{
+			Name:      "installer-cloud-credentials",
+			Namespace: "openshift-image-registry",
+			RoleARN:   rosaScope.ControlPlane.Spec.RolesRef.ImageRegistryARN,
+		},
+		{
+			Name:      "ebs-cloud-credentials",
+			Namespace: "openshift-cluster-csi-drivers",
+			RoleARN:   rosaScope.ControlPlane.Spec.RolesRef.StorageARN,
+		},
+		{
+			Name:      "cloud-credentials",
+			Namespace: "openshift-cloud-network-config-controller",
+			RoleARN:   rosaScope.ControlPlane.Spec.RolesRef.NetworkARN,
+		},
+		{
+			Name:      "kube-controller-manager",
+			Namespace: "kube-system",
+			RoleARN:   rosaScope.ControlPlane.Spec.RolesRef.KubeCloudControllerARN,
+		},
+		{
+			Name:      "kms-provider",
+			Namespace: "kube-system",
+			RoleARN:   rosaScope.ControlPlane.Spec.RolesRef.KMSProviderARN,
+		},
+		{
+			Name:      "control-plane-operator",
+			Namespace: "kube-system",
+			RoleARN:   rosaScope.ControlPlane.Spec.RolesRef.ControlPlaneOperatorARN,
+		},
+		{
+			Name:      "capa-controller-manager",
+			Namespace: "kube-system",
+			RoleARN:   rosaScope.ControlPlane.Spec.RolesRef.NodePoolManagementARN,
+		},
+	} {
+		roles = append(roles, cmv1.NewOperatorIAMRole().
+			Name(role.Name).
+			Namespace(role.Namespace).
+			RoleARN(role.RoleARN))
+	}
+	stsBuilder = stsBuilder.OperatorIAMRoles(roles...)
+
+	instanceIAMRolesBuilder := cmv1.NewInstanceIAMRoles()
+	instanceIAMRolesBuilder.MasterRoleARN("TODO")
+	instanceIAMRolesBuilder.WorkerRoleARN("TODO")
+	stsBuilder = stsBuilder.InstanceIAMRoles(instanceIAMRolesBuilder)
+	stsBuilder.OidcConfig(cmv1.NewOidcConfig().ID(*rosaScope.ControlPlane.Spec.OIDCID))
+
+	stsBuilder.AutoMode(true)
+
+	awsBuilder := cmv1.NewAWS().
+		AccountID(*rosaScope.ControlPlane.Spec.AccountID)
+	awsBuilder = awsBuilder.SubnetIDs(rosaScope.ControlPlane.Spec.Subnets...)
+	awsBuilder = awsBuilder.STS(stsBuilder)
+	clusterBuilder = clusterBuilder.AWS(awsBuilder)
+
+	clusterNodesBuilder := cmv1.NewClusterNodes()
+	clusterNodesBuilder = clusterNodesBuilder.AvailabilityZones("us-west-2a")
+	clusterBuilder = clusterBuilder.Nodes(clusterNodesBuilder)
+
+	clusterProperties := map[string]string{}
+	clusterProperties["rosa_creator_arn"] = *rosaScope.ControlPlane.Spec.CreatorARN
+
+	clusterBuilder = clusterBuilder.Properties(clusterProperties)
+	clusterSpec, err := clusterBuilder.Build()
+	if err != nil {
+		return reconcile.Result{}, fmt.Errorf("failed to create description of cluster: %v", err)
+	}
+
+	// Create a logger that has the debug level enabled:
+	ocmLogger, err := sdk.NewGoLoggerBuilder().
+		Debug(true).
+		Build()
+	if err != nil {
+		return reconcile.Result{}, fmt.Errorf("failed to build logger: %w", err)
+	}
+
+	// Create the connection, and remember to close it:
+	token := os.Getenv("OCM_TOKEN")
```

Comment:
This means that the users will have to modify the CAPA deployment. Could the token come from a secret?

Reply:
yes, the token would be used for all clusters.

I agree the way it is right now is not great. We will need to find a better way to pass this token, a secret could be good solution.

I will create an issue to track that.

Reply:
Can you please leave the issue link in the comment above this so we know this is being thought about?